### ISSUE TITLE
feat: Full entrenar distillation integration

### ISSUE THREAD

**Issue by noahgift** [paiml/verificar#4]
## Summary
Integrate entrenar's knowledge distillation module with verificar's codex pipeline.

## entrenar Distillation API
- `DistillationLoss` - temperature-scaled KL divergence + cross-entropy
- `EnsembleDistiller` - multi-teacher ensemble distillation  
- `ProgressiveDistiller` - layer-wise distillation

## Requirements
- [ ] Load teacher logits from depyler-oracle or external LLM
- [ ] Support multi-teacher ensemble (depyler + ruchy + bashrs oracles)
- [ ] Generate student model architecture config
- [ ] Integrate with `verificar codex` as Stage 4

## CLI Usage
```bash
verificar distill \
  --input teacher_logits/ \
  --output distilled_model/ \
  --temperature 3.0 \
  --alpha 0.7 \
  --num-teachers 3 \
  --epochs 10
```

## References
- entrenar `src/distill/` module
- Hinton et al. (2015) "Distilling the Knowledge in a Neural Network"
